build/musicbrainz-dev/scripts/fetch-dump.sh

```diff
@@ -98,18 +98,37 @@ then
 	exit 64 # EX_USAGE
 fi
 
+# Keep support for (deprecated) FTP option (which still takes precedence)
+
+BASE_DOWNLOAD_URL="${BASE_FTP_URL:-$BASE_DOWNLOAD_URL}"
+
 # Fetch latest search indexes
 
 if [[ $TARGET =~ ^(both|search)$ ]]
 then
 	echo "$(date): Fetching search indexes dump..."
-	cd "$SEARCH_DUMP_DIR" && find . -delete && cd -
+	PREVIOUS_DUMP_TIMESTAMP=$(
+		if [[ -a "$SEARCH_DUMP_DIR/LATEST" ]]
+		then
+			cat "$SEARCH_DUMP_DIR/LATEST"
+		fi
+	)
+	rm -f "$SEARCH_DUMP_DIR/LATEST"
 	"${WGET_CMD[@]}" -nd -nH -P "$SEARCH_DUMP_DIR" \
-		"${BASE_FTP_URL:-$BASE_DOWNLOAD_URL}/data/search-indexes/LATEST"
-	DUMP_TIMESTAMP=$(cat /media/searchdump/LATEST)
-	"${WGET_CMD[@]}" -nd -nH -r -P "$SEARCH_DUMP_DIR" \
-		"${BASE_FTP_URL:-$BASE_DOWNLOAD_URL}/data/search-indexes/$DUMP_TIMESTAMP/"
-	cd "$SEARCH_DUMP_DIR" && md5sum -c MD5SUMS && cd -
+		"${BASE_DOWNLOAD_URL}/data/search-indexes/LATEST"
+	DUMP_TIMESTAMP=$(cat "$SEARCH_DUMP_DIR/LATEST")
+	if [[ $PREVIOUS_DUMP_TIMESTAMP != "$DUMP_TIMESTAMP" ]]
+	then
+		find "$SEARCH_DUMP_DIR" \
+			! -path "$SEARCH_DUMP_DIR" \
+			! -path "$SEARCH_DUMP_DIR/LATEST" \
+			-delete
+	fi
+	"${WGET_CMD[@]}" -nd -nH -c -r -P "$SEARCH_DUMP_DIR" \
+		--accept 'MD5SUMS,*.tar.zst' --no-parent --relative \
```

Comment:
Could we use `SHA256SUMS` instead? Or are there some compatibility issues I'm not aware of?

Reply:
Yes, we could. The command `sha256sum` is even installed in our base image.

It is slightly slower though, for not much security improvement given that the checksum file is downloaded using HTTP by default and that its signature is not verified.

Ideally, we should verify the signatures and use SHA. It seems worth a separate ticket/PR.

Reply:
Thanks, definitely something for a separate PR.